src/FinSearch.php

```diff
@@ -79,11 +99,52 @@ private function loadServiceXml($container, string $filePath): void
         $loader->load('services.xml');
     }
 
+    private function isCompatible(InstallContext $installContext): bool
```

Comment:
Make this method public and static. This way we can properly unit-test it. Add the composer.json destination as optional parameter and override it in the tests. Ensure that

* composer.json contains `^6.1||^6.2||^6.3`
  * Shopware 6.1.0, 6.1.6, 6.2.0, 6.2.3, 6.3.0.0, 6.3.3.0 and 6.3.5.1 is compatible.
  * Shopware 6.4.0.0, 6.4.5.0, 6.5.1.2, 6.7.6.4 is not compatible.

This way we have this properly tested and can ensure that this works as expected.